assets/js/components/GoogleChart/utils.js

```diff
@@ -0,0 +1,200 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import set from 'lodash/set';
+import cloneDeep from 'lodash/cloneDeep';
+import merge from 'lodash/merge';
+
+/**
+ * Internal dependencies
+ */
+import { stringToDate } from '../../util/date-range/string-to-date';
+
+/**
+ * Returns the Google chart data, filtered by selected stats if present.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array|undefined} data          Chart data.
+ * @param {Array|undefined} selectedStats The columns that should be displayed for the data set.
+ * @return {Object} The chart data, filtered by selected stats if present.
+ */
+export const getFilteredChartData = ( data, selectedStats ) => {
+	if ( ! selectedStats?.length ) {
+		return data;
+	}
+	// Ensure we don't filter out columns that aren't data, but are things like
+	// tooltips or other content.
+	let nonDataColumns = [];
+	if ( data?.length ) {
+		nonDataColumns = data[ 0 ].reduce( ( acc, row, rowIndex ) => {
+			return row?.role ? [ ...acc, rowIndex ] : acc;
+		}, [] );
+	}
+
+	let filteredData = data;
+	if ( selectedStats?.length > 0 ) {
+		filteredData = data.map( ( row ) => {
+			return row.filter( ( _columnValue, columnIndex ) => {
+				return (
+					columnIndex === 0 ||
+					selectedStats.includes( columnIndex - 1 ) ||
+					nonDataColumns.includes( columnIndex - 1 )
+				);
+			} );
+		} );
+	}
+
+	return filteredData;
+};
+
+/**
+ * Returns the optimal height and width for a preview element.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string|undefined} loadingHeight Preview height.
+ * @param {string|undefined} height        Chart height.
+ * @param {string|undefined} loadingWidth  Preview width.
+ * @param {string|undefined} width         Chart width.
+ * @return {Object} The optimal height and width to use in a preview element.
+ */
+export const getLoadingDimensions = (
+	loadingHeight,
+	height,
+	loadingWidth,
+	width
+) => {
+	const dimensions = {
+		height: loadingHeight || height,
+		width: loadingWidth || width,
+	};
+	// If a loading height is set but a width is not (or a loading width is set
+	// but not a height), change the "unset" value to 100% to avoid visual bugs.
+	// See: https://github.com/google/site-kit-wp/pull/2916#discussion_r623866269
+	if ( dimensions.width && ! dimensions.height ) {
+		dimensions.width = '100%';
+	}
+	if ( dimensions.height && ! dimensions.width ) {
+		dimensions.height = '100%';
```

Comment:
`width` and `height` need to be switched here as they should only be set when not already set.

```suggestion
		dimensions.height = '100%';
	}
	if ( dimensions.height && ! dimensions.width ) {
		dimensions.width = '100%';
```